code/controllers/subsystem/dbcore.dm

```diff
@@ -0,0 +1,437 @@
+	flags = SS_BACKGROUND
+	wait = 1 MINUTES
+	init_order = INIT_ORDER_DBCORE
+
+	/// Is the DB schema valid
+	var/schema_valid = TRUE
+	/// Timeout of failed connections
+	var/failed_connection_timeout = 0
+	/// Amount of failed connections
+	var/failed_connections = 0
+
+	/// Last error to occur
+	var/last_error
+	/// List of currenty processing queries
+	var/list/active_queries = list()
+
+	/// SQL errors that have occured mid round
+	var/total_errors = 0
+
+	/// Connection handle. This is an arbitrary handle returned from rust_g.
+	var/connection
+
+	offline_implications = "The server will no longer check for undeleted SQL Queries. No immediate action is needed."
+
+/datum/controller/subsystem/dbcore/stat_entry()
+	..("A: [length(active_queries)]")
+
+// This is in Initialize() so that its actually seen in chat
+/datum/controller/subsystem/dbcore/Initialize()
+	if(!schema_valid)
+		to_chat(world, "<span class='boldannounce'>Database schema ([sql_version]) doesn't match the latest schema version ([SQL_VERSION]). Roundstart has been delayed.</span>")
+
+	return ..()
+
+/datum/controller/subsystem/dbcore/fire()
+	for(var/I in active_queries)
+		var/datum/db_query/Q = I
+		if(world.time - Q.last_activity_time > (5 MINUTES))
+			message_admins("Found undeleted query, please check the server logs and notify coders.")
+			log_sql("Undeleted query: \"[Q.sql]\" LA: [Q.last_activity] LAT: [Q.last_activity_time]")
+			qdel(Q)
+		if(MC_TICK_CHECK)
+			return
+
+/datum/controller/subsystem/dbcore/Recover()
+	connection = SSdbcore.connection
+
+//nu
+/datum/controller/subsystem/dbcore/can_vv_get(var_name)
+	return var_name != NAMEOF(src, connection) && var_name != NAMEOF(src, active_queries) && ..()
+
+/datum/controller/subsystem/dbcore/vv_edit_var(var_name, var_value)
+	if(var_name == NAMEOF(src, connection))
+		return FALSE
+	return ..()
+
+/**
+  * Connection Creator
+  *
+  * This proc basically does a few sanity checks before connecting, then attempts to make a connection
+  * When connecting, RUST_G will initialize a thread pool for queries to use to run asynchronously
+  */
+/datum/controller/subsystem/dbcore/proc/Connect()
+	if(IsConnected())
+		return TRUE
+
+	if(failed_connection_timeout <= world.time) //it's been more than 5 seconds since we failed to connect, reset the counter
+		failed_connections = 0
+
+	if(failed_connections > 5)	//If it failed to establish a connection more than 5 times in a row, don't bother attempting to connect for 5 seconds.
+		failed_connection_timeout = world.time + 50
+		return FALSE
+
+	if(!config.sql_enabled)
+		return FALSE
+
+	var/result = json_decode(rustg_sql_connect_pool(json_encode(list(
+		"host" = sqladdress,
+		"port" = text2num(sqlport),
+		"user" = sqlfdbklogin,
+		"pass" = sqlfdbkpass,
+		"db_name" = sqlfdbkdb,
+		"read_timeout" = config.async_sql_query_timeout,
+		"write_timeout" = config.async_sql_query_timeout,
+		"max_threads" = config.rust_sql_thread_limit,
+	))))
+	. = (result["status"] == "ok")
```

Comment:
```suggestion
	. = result["status"] == "ok"
```

Reply:
Disagree, this helps with styling and makes it much more obvious that its a bool